src/hotspot/share/runtime/handshake.cpp

```diff
@@ -481,12 +480,14 @@ bool HandshakeState::process_self_inner() {
       if (!async) {
         HandleMark hm(_handshakee);
         PreserveExceptionMark pem(_handshakee);
-        op->do_handshake(_handshakee);
+        op->do_handshake(_handshakee); // acquire, op removed after
```

Comment:
Why `acquire`? Perhaps:
`// peek and handshake with op; op removed after`

Reply:
It's mean do_handshake provide acquire thus op is removed after.

Reply:
Ok.